sklearn/model_selection/tests/test_validation.py

```diff
@@ -2622,6 +2624,37 @@ def test_cross_validate_routing(cv_method):
             metadata=fit_metadata,
         )
 
+    # learning_curve uses an internal function for partial fitting the estimator when
+    # exploit_incremental_learning=True
+
+    estimator_registry = _Registry()
+    estimator = ConsumingClassifier(
+        registry=estimator_registry
+    ).set_partial_fit_request(
+        sample_weight="fit_sample_weight", metadata="fit_metadata"
+    )
+
+    if func is learning_curve:
```

Comment:
I would move this to a separate test instead of adding to this test.

Reply:
Done.